### ISSUE TITLE
explore using Prometheus metrics to report on usage, users, and cloud costs with Jupyter notebooks for this hub

### ISSUE THREAD

**Issue by jmunroe** [2i2c-org/community-showcase#21]

**Comment by jmunroe** [comment]
@choldgraf made this [comment](https://2i2c.slack.com/archives/C055A1J1DRP/p1698856026246659?thread_ts=1698855479.057689&cid=C055A1J1DRP) in Slack about using Prometheus/Grafana to understand the usage of our infrastructure

> mostly I've wanted to use those graphs as a source of truth for our usage, number of hubs, etc, but it's too confusing for me to be able to use

In this sprint, I'd like to iterate on why this difficult and come up with a plan for what can be done today, and what we need to plan for the future development effort.

**Comment by jmunroe** [comment]
@sgibson91 -- I am considering asking questions about Grafana as part of this sprint.  Is this a topic you'd be able to help me understand better? (Even if you are not an expert, just having the conversation around what is expected and what should not be expected would be valuable).

**Comment by choldgraf** [comment]
Just cross-linking the latest ask I made around the Global Usage Dashboard for Grafana:

https://github.com/2i2c-org/infrastructure/issues/1888#issuecomment-1432703512

And for reference, here's a short video showing how to create the graph I _want_ (more or less). I don't think it'd be that hard to do it, but somebody just needs to make the proper adjustments.

https://github.com/2i2c-org/community-showcase/assets/1839645/5d2d883f-6ef0-4b57-a3e8-1f21497b6163

**Comment by sgibson91** [comment]
> Can new Grafana dashboards be created?

Yes. The most robust way is to contribute them to this project: https://github.com/jupyterhub/grafana-dashboards
Once it is merged upstream, we can run our workflow here to deploy the dashboard to all our hubs: https://github.com/2i2c-org/infrastructure/actions/workflows/deploy-grafana-dashboards.yaml (This action overwrites any dashboards that were manually created in the grafana interface.)

This about covers the extent of my grafana knowledge.

**Comment by jmunroe** [comment]
No worries about grafana knowledge. It's an area that I keep on hitting as something I wish I knew more about. I'll pull you off this issue for the current sprint and let's prioritize other tasks first!

**Comment by jmunroe** [comment]
@yuvipanda has written up some notes get me started: https://hackmd.io/NllqOUfaTLCXcDQPipr4rg
- https://github.com/2i2c-org/infrastructure/issues/3426

**Comment by colliand** [comment]
Recently, @yuvipanda wrote up a [HackMD note "Access promethues data raw"](https://hackmd.io/NllqOUfaTLCXcDQPipr4rg) in response to a request from @colliand. Yuvi's work should enable 2i2c's Parterships team to build notebook(s) that enable hub champions to get insights into hub usage patterns.

**Comment by yuvipanda** [comment]
Note that I got the same request from 3 people (@jmunroe, @colliand and Juliusz from LEAP) on the exact same week by coincidence :) https://github.com/2i2c-org/infrastructure/issues/3426 tracks the documentation itself, and will figure out where it lands.

**Comment by damianavila** [comment]
@jmunroe, @sgibson91, there is a chance to synergistically work with @GeorgianaElena (and I presume Yuvi) because she wants to move forward with some discussions/clarifications/docs/fixes about Grafana as part of the "Reduce Workload" goal last sprint starting tomorrow.
